src/sources.js

```diff
@@ -294,13 +294,18 @@ class S3Source extends Source {
     return `https://${this.bucket}.s3.amazonaws.com`;
   }
   async _listObjects() {
-    // XXX TODO: This will only return the first 1000 objects.  That's fine for
-    // now (for comparison, nextstrain-data only has ~500), but we really
-    // should iterate over the whole bucket contents using the S3 client's
-    // pagination support.
-    //   -trs, 30 Aug 2019
-    const list = await S3.listObjectsV2({Bucket: this.bucket}).promise();
-    return list.Contents;
+    const s3Params = {Bucket: this.bucket, MaxKeys: 1000};
+    let contents = [];
+    let existMoreKeys = true;
+    let lastKeySeen = false;
```

Comment:
You don't need to assign a value here; the initialized variable `lastKeySeen` will evaluate as a falsy boolean.
e.g.
```
let lastKeySeen
!!lastKeySeen
> false
```

Reply:
This is no longer needed in the new implementation using `eachPage` but I did this on purpose as I preferred the explicitness of it :)

Reply:
Ditto re: my calling it a nitpick. I completely understand the desire to be more explicit.